delivery_carrier_option/models/delivery_carrier_template_option.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2012 Akretion <http://www.akretion.com>.
+# Copyright 2013-2016 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class DeliveryCarrierTemplateOption(models.Model):
+    """Available options for a carrier (partner)"""
+
+    _name = "delivery.carrier.template.option"
+    _description = "Delivery carrier template option"
+
+    type = fields.Selection(
+        selection=[
+            ("basic", "Basic Service"),
+            ("additional", "Additional Service"),
+            ("delivery", "Delivery Instructions"),
+            ("partner_option", "Partner Option"),
+        ],
+        default="basic",
+        string="Option type",
+    )
+    partner_id = fields.Many2one(comodel_name="res.partner", string="Partner Carrier")
+    name = fields.Char(readonly=True)
+    code = fields.Char(readonly=True)
+    description = fields.Char(
+        readonly=True,
+        help="Allow to define a more complete description " "than in the name field.",
+    )
```

Comment:
```suggestion
    name = fields.Char()
    code = fields.Char()
    description = fields.Char(
        help="Allow to define a more complete description " "than in the name field.",
    )
```

Would it makes sense to move the `readonly` attribute to the views? 

> If a field is marked as readonly=True on the model, it won't be possible to import it through the Odoo import tool, so avoid it as possible and define the readonly attribute on the views instead.

https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0

Reply:
I don't know, do you need to import it ?
The template options have to be integrated with the carrier api, so it should be part of  the specific carrier module implementation, and not be created by import right ?

Anyway, if someone using this module needs it, I am ok to change it , but if it is not the case, I would rather not.

Reply:
These options could be imported on the fly. So it is better to restrict on the view instead on the model.